requests_credssp/credssp.py

```diff
@@ -309,15 +310,55 @@ def _send_encrypted_credentials(self, response, **kwargs):
         encrypted_credential, encrypted_credential_sig = self.context.session_security.wrap(ts_credentials.get_data())
         credential_ts_request['auth_info'].value = encrypted_credential_sig + encrypted_credential
 
-        self.tls_connection.send(credential_ts_request.get_data())
-        credential_credssp_token = self.tls_connection.bio_read(8192)
+        credential_credssp_token = self.wrap(credential_ts_request.get_data())
         request = response.request.copy()
         self._set_credssp_token(request, credential_credssp_token)
 
         response = response.connection.send(request, **kwargs)
 
         return response
 
+    def wrap(self, data):
+        """
+        Encrypts the data in preparation for sending to the server. The data is
+        encrypted using the TLS channel negotiated between the client and the
+        server.
+
+        :param data: a byte string of data to encrypt
+        :return: a byte string of the encrypted data
+        """
+        self.tls_connection.send(data)
+        encrypted_data = b''
+
+        while True:
+            try:
```

Comment:
Should probably have a sanity counter w/ forced exit on both of these loops around some multiple of len(data)- if the TLS session is borked without throwing (seen it happen, sadly), we'd technically go forever on this.